store/task.go

```diff
@@ -317,18 +317,20 @@ func (s *TaskService) findTaskList(ctx context.Context, tx *sql.Tx, find *api.Ta
 		taskRunFind := &api.TaskRunFind{
 			TaskID: &task.ID,
 		}
-		task.TaskRunList, err = s.TaskRunService.FindTaskRunListTx(ctx, tx, taskRunFind)
+		taskRunList, err := s.TaskRunService.FindTaskRunListTx(ctx, tx, taskRunFind)
 		if err != nil {
 			return nil, err
 		}
+		task.TaskRunList = taskRunList
 
 		taskCheckRunFind := &api.TaskCheckRunFind{
 			TaskID: &task.ID,
 		}
-		task.TaskCheckRunList, err = s.TaskCheckRunService.FindTaskCheckRunListTx(ctx, tx, taskCheckRunFind)
+		taskCheckRunList, err := s.TaskCheckRunService.FindTaskCheckRunListTx(ctx, tx, taskCheckRunFind)
```

Comment:
I am wondering the benefit of returning nil instead of returning [] for the list operation.

Also is it just a safe measure to use a temp variable? For this example (and I think in other places as well), we will return err, thus whether setting task.TaskCheckRunList does not matter. We just opt to write in a terse way.

Reply:
According to Go code review wiki, it's more preferable to declare empty slices right away, i.e. nil is preferred over empty slice. Same for return value.
  https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
> thus whether setting task.TaskCheckRunList does not matter
It's a tradeoff between one line less code & no side effect. I prefer the latter.

Reply:
And there's one good thing about returning nil in the composition process: we could always check the pointers in some recovery function to tell whether one relationship is composed or not.

Reply:
@tianzhou @d-bytebase Are we agreed or not on this pattern?